### ISSUE TITLE
Item sorting and resource routing to storage.

### ISSUE THREAD

**Issue by LadyNasse** [StrangeLoopGames/EcoSuggestions#1327]
TL;DR: It would be really nice to be able to filter items in a view, i.e. set it to only show specific items, or item tags, making it easier to look through different linked inventories. It would ALSO be nice to "gate" storages, i.e. marking stockpiles or storage boxes and the like to only accept certain items or item tags.

Long version, for clarity: When looking at a table or storage and then viewing the linked storages, it would be really nice to be able to click checkboxes on a side menu (much like the chat's collapsible menu) to filter items by name or tag. I.E. selecting "sandstone" to view all items with the word sandstone in the name, or "sandstone item" to view JUST sandstone rocks, or "rock tag" to view all rocks regardless of type. It would ALSO be really nice to create custom search groups, for frequent use. Like maybe I always need to find 2, 3, or 4 items. I could create a custom filter and then add it to my menu and select it easily later on.

Total side note, it would ALSO be really nice if, when placing a table or storage, if there is already the same type of it already on the deed, it would automatically add a numerical iteration to the end of the. I.E. placing a second workbench on my deed and having it automatically be "Lady Nasse's Workbench 2" or something. In this way when placing multiple stockpiles or storage chests and whatnot, you can obviously rename them if you want, but otherwise you just have a quick and dirty identifier from go.

As for the stockpile gating... As an example say that I have a work table set up with stockpiles arranged so it outputs to stockpile A, first, and then Stockpile B second. Now let's say I have some work orders I would like to output to Stockpile A, so that's fine, but other times I want some of the orders to go to Stockpile B, instead. As the game stands the only way to accomplish this is to have different tables with their linked storages organized differently and then do the work orders table specific, or swap the order of the linked stockpiles between jobs, which is obviously cumbersome. A better method, imo, would be to either gate the storages to only accept specific input types, either by exact item names, or by item tags... or be able to select a storage output option on each work order, as an optional setting. I'd argue that second option is more versatile, especially when doing work contracts and contributing labor vs starting a work order.

**Comment by sirPattrick** [comment]
This is a huge must.   When you have 6 Storage chests and 3 stockpiles, it would be nice to be able to organize them and sort them.  The above being able to filter, and also the ability to organize within the selected Storages would be a huge win (and a button to auto deposit all items already in the storage.  similar to what is in StarDew Valley.)
